sklearn/linear_model/least_angle.py

```diff
@@ -48,22 +47,266 @@ def lars_path(X, y, Xy=None, Gram=None, max_iter=500,
 
     Parameters
     -----------
-    X : array, shape: (n_samples, n_features)
-        Input data.
+    X : None or array, shape (n_samples, n_features)
+        Input data. Note that if X is None then the Gram matrix must be
+        specified, i.e., cannot be None or False.
 
-    y : array, shape: (n_samples)
+    y : None or array, shape (n_samples,)
+        Input targets.
+
+    Xy : array-like, shape (n_samples,) or (n_samples, n_targets), optional
+        Xy = np.dot(X.T, y) that can be precomputed. It is useful
+        only when the Gram matrix is precomputed.
```

Comment:
If this is deprecated, we should say so here. Use sphinx's `.. deprecated`

Reply:
Fixed.